src/app/public/modules/character-counter/character-counter-indicator.component.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  OnInit
+} from '@angular/core';
+
+import {
+  SkyLibResourcesService
+} from '@skyux/i18n';
+
+import {
+  Subject
+} from 'rxjs/Subject';
+
+@Component({
+  selector: 'sky-character-counter-indicator',
+  templateUrl: './character-counter-indicator.component.html',
+  styleUrls: ['./character-counter-indicator.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyCharacterCounterIndicatorComponent implements OnInit {
+
+  public currentCharacterCountLimit: number = 0;
+  public currentCharacterCount: number = 0;
+  public characterCountMessage: string;
+  public overLimitMessage: string;
+  private ngUnsubscribe = new Subject<void>();
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private libResources: SkyLibResourcesService
+    ) { }
+
+  public ngOnInit(): void {
+    this.getCharacterCountMessage();
+  }
+
+  public get characterCount(): number {
+    return this.currentCharacterCount;
+  }
+
+  public set characterCount(count: number) {
+    this.currentCharacterCount = count;
+    this.getCharacterCountMessage();
+    this.changeDetector.markForCheck();
+  }
+
+  public get characterCountLimit(): number {
+    return this.currentCharacterCountLimit;
+  }
+
+  public set characterCountLimit(limit: number) {
+    this.currentCharacterCountLimit = limit;
+    this.getCharacterCountMessage();
+    this.changeDetector.markForCheck();
+  }
+
+  public getCharacterCountMessage(): void {
```

Comment:
Methods (and getters) that are used in the component's template get called indiscriminately during every change detection cycle. You can test this out for yourself by putting a console log in the `getCharacterCountMessage` method and seeing how many times it gets called when interacting with the component during a serve.

Also, it appears that we're creating a new subscription every time the `getCharacterCountMessage` is hit.

Is there a way to call `SkyLibResources.getString` method less frequently, and maybe manage the number of subscriptions a bit better? Maybe you could look into using the `SkyAppFormat` utility for each subsequent change detection, after initially retrieving the resource string from `SkyLibResources`. See: https://github.com/blackbaud/skyux-core/blob/master/src/app/public/modules/format/app-format.ts

Reply:
Another less important aspect to think about is the method name. The word "get" implies that something would be returned from the method call.